### DISCUSSION TITLE
Get HTTP response status code on fetchers

### DISCUSSION THREAD

**Discussion by ZeldOcarina** [remix-run/remix#8147] [Q&A]
I have a question, how would I get the response status code from a fetcher's action request?
I only get back the response data.

Right now I am manually adding in the response JSON the status as part of the response data but I think there should be a better way to access if the response was 400, 201, 200 etc.

Thanks for the attention and for mantaining this beautiful framework.

**Comment by sergiodxa** [comment]
There’s no way. You need to use the response data to know if it was a success or not.

**Reply by gustavopch** [reply]
Adding to Sergio's answer, might be better to use your own descriptive statuses instead of HTTP's. Like, if you run an action to update a product and it wasn't found, you could return `{ status: 'product-not-found' }` instead of `{ status: 404 }`.

**Reply by ZeldOcarina** [reply]
Yeah I am doing it but it seems a bit strange to not be able to access that, all other tools usually expose it, things like xhr, fetch, axios etc.. is this a specific design decision or is it something we may discuss to include in the fetcher's response in the future?

**Reply by sergiodxa** [reply]
> Adding to Sergio's answer, might be better to use your own descriptive statuses instead of HTTP's. Like, if you run an action to update a product and it wasn't found, you could return `{ status: 'product-not-found' }` instead of `{ status: 404 }`.

💯 agree, I have started to do something like this

```ts
return json({
  intent: "the form intent" as const,
  status: "error" as const, // or `"success" as const`
  errors: { ... } // the error messages for each field
})
```

And for success I set the `intent`, the `status` as success and if needed a `data`. And then TS let me use both intent and status to get type safety.

**Comment by ZeldOcarina** [comment]
@ryanflorence I'd love your input on this if it wouldn't bother you.

**Comment by bzitzow-osh** [comment]
> There’s no way. You need to use the response data to know if it was a success or not.

Given the RemixJS philosophy of utilizing the web platform as much as possible - the fact that this is the recommendation  _is wildly surprising to me_

The answer is essentially:
* Don't use HTTP
* Roll your own

Which seems antithetical to the philosophy.

**Reply by ZeldOcarina** [reply]
Yeah this makes zero sense, can we make this into an issue @ryanflorence ?

**Reply by b-goldney** [reply]
Was there ever additional insight into this? Agreed that it seems antithetical to the philosophy.

**Comment by devbytyler** [comment]
Stumbled upon this today. Feels like an oversight, but I am sure it was intentional. Would love to know the reasoning behind this. Same with `useActionData` -- no obvious way to get the status.

**Reply by ZeldOcarina** [reply]
@ryanflorence
